src/hooks/spaces/useSpaceBySlug.ts

```diff
@@ -0,0 +1,65 @@
+import { useFirestore, useFirestoreCollectionData } from "reactfire";
+import Bugsnag from "@bugsnag/js";
+
+import { AnyVenue } from "types/venues";
+
+import { withIdConverter } from "utils/converters";
+import { WithId } from "utils/id";
+
+export type UseSpaceBySlugResult = {
+  space?: WithId<AnyVenue>;
+  isLoaded: boolean;
+  error?: string;
+};
+
+/**
+ * Hook which will return the space when the slug is provided.
+ * The intention is to be used on the client side, when the space slug is provided in the url.
+ * @param spaceSlug
+ * @returns
+ */
+export const useSpaceBySlug: (spaceSlug?: string) => UseSpaceBySlugResult = (
+  spaceSlug
+) => {
+  const firestore = useFirestore();
+
+  const spacesRef = firestore
+    .collection("venues")
+    .where("slug", "==", spaceSlug ?? "")
+    .withConverter(withIdConverter<AnyVenue>());
+
+  const { data: spaces, status } = useFirestoreCollectionData<WithId<AnyVenue>>(
```

Comment:
I guess we can go with `data: spaces = []`, so that we can create smth like `const [firstSpace] = spaces` and not repeat ourselves with `!spaces?.[0]` down below. But it might be too much tho

Reply:
You do have a point, there are however a couple of props that accept `venueId` and I won't go that deep to change those as well.

Reply:
It is no problem the `useSpaceBySlug()` hook to return few more items for greater flexibility/robustness, like:
`venueId`, `spaceId`, `worldId` 
alongside `space`, `venue` 
even (and why not?) an `isLoaded` and `isNotLoaded`.

Some of them can be used, others can be marked as deprecated until their usages are removed and dealt with them at that point.